src/components/NoEntityCard/NoEntityCard.test.tsx

```diff
@@ -0,0 +1,42 @@
+import { Button } from "@canonical/react-components";
+import { act, screen, within } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { vi } from "vitest";
+
+import { renderComponent } from "test/utils";
+
+import NoEntityCard from "./NoEntityCard";
+
+const title = "Mock title";
+const message = "Mock message";
+
+test("should render correctly without action button", async () => {
+  renderComponent(<NoEntityCard title={title} message={message} />);
+  const emptyStateHeader = await screen.findByRole("heading", { name: title });
+  const emptyState = emptyStateHeader.parentElement;
+  expect(emptyState).not.toBeNull();
+  expect(within(emptyState!).getByText(message)).toBeInTheDocument();
```

Comment:
This should probably use the test id rather parentElement to avoid the non-null assertion, though I'm not sure if the `within()` is necessary here. I think the only way that that would fail is if somehow we moved the message outside of the wrapping element, but I'm not sure that's a failure mode that needs a test.